cosmos/models/Workflow.py

```diff
@@ -443,6 +446,19 @@ def delete(self, delete_files=False):
         self.session.commit()
         print >> sys.stderr, '%s Deleted' % self
 
+    def exit_status(self):
+        """
+        Return the exit status of the first failed Task (topologically, not chronologically).
+
+        If no Task failed, return None.
+        """
+        for t in self.task_graph():
+            if t.exit_status:
+                self.log.warning("%s took exit status %s from %s (%s)",
+                                 self, t.exit_status, t, t.status)
+                return t.exit_status
+        return None
+
```

Comment:
I don't think you're being too nit picky at all. Not having use `nx.DiGraph()` before, I mistakenly assumed it iterated in topological order. Thanks for the review and the suggestion.